File: c/10_ssrf.c
/* A10 - SSRF-ish in native code. VULN: curl to user URL without validation. */
#include <stdio.h>
#include <stdlib.h>

/* VULN: A10 - passes URL to curl via system() */
void fetch_url(const char *url) {
  char cmd[512];
  snprintf(cmd, sizeof(cmd), "curl -s '%s'", url);
  system(cmd); /* command injection + SSRF */
}

File: go/10_ssrf.go
// A10 - SSRF. VULN: http.Get on user-controlled URL.
package main

import (
	"io"
	"net/http"
)

// VULN: A10
func FetchPreview(url string) ([]byte, error) {
	resp, err := http.Get(url)
	if err != nil {
		return nil, err
	}
	defer resp.Body.Close()
	return io.ReadAll(resp.Body)
}
